packages/flutter/test/widgets/focus_scope_test.dart

```diff
@@ -0,0 +1,676 @@
+  testWidgets('Can have multiple focused children and they update accordingly', (WidgetTester tester) async {
+    final GlobalKey<TestFocusableState> keyA = GlobalKey();
+    final GlobalKey<TestFocusableState> keyB = GlobalKey();
+
+    await tester.pumpWidget(
+      Column(
+        children: <Widget>[
+          TestFocusable(
+            key: keyA,
+            name: 'a',
+            autofocus: true,
+          ),
+          TestFocusable(
+            key: keyB,
+            name: 'b',
+          ),
+        ],
+      ),
+    );
+
+    // Autofocus is delayed one frame.
+    await tester.pump();
+
+    expect(keyA.currentState.focusNode.hasFocus, isTrue);
+    expect(keyB.currentState.focusNode.hasFocus, isFalse);
+    await tester.idle();
+    await tester.pump();
+    expect(keyA.currentState.focusNode.hasFocus, isTrue);
+    expect(keyB.currentState.focusNode.hasFocus, isFalse);
+    await tester.tap(find.text('A FOCUSED'));
+    await tester.idle();
+    await tester.pump();
+    expect(keyA.currentState.focusNode.hasFocus, isTrue);
+    expect(keyB.currentState.focusNode.hasFocus, isFalse);
+    await tester.tap(find.text('b'));
+    await tester.idle();
+    await tester.pump();
+    expect(keyA.currentState.focusNode.hasFocus, isFalse);
+    expect(keyB.currentState.focusNode.hasFocus, isTrue);
+    await tester.tap(find.text('a'));
+    await tester.idle();
+    await tester.pump();
+    expect(keyA.currentState.focusNode.hasFocus, isTrue);
+    expect(keyB.currentState.focusNode.hasFocus, isFalse);
+  });
+
+  testWidgets('Can move focus in and out of FocusScope', (WidgetTester tester) async {
+    final FocusScopeNode parentFocusScope = FocusScopeNode();
+    final FocusScopeNode childFocusScope = FocusScopeNode();
+    final GlobalKey<TestFocusableState> key = GlobalKey();
+
+    await tester.pumpWidget(
+      FocusScope(
+        node: parentFocusScope,
+        autofocus: true,
+        child: Column(
+          children: <Widget>[
+            TestFocusable(key: key),
+          ],
+        ),
+      ),
+    );
+
+    expect(key.currentState.focusNode.hasFocus, isFalse);
+    FocusScope.of(key.currentContext).requestFocus(key.currentState.focusNode);
+    await tester.idle();
+    await tester.pump();
+
+    expect(key.currentState.focusNode.hasFocus, isTrue);
+
+    expect(parentFocusScope, hasAGoodToStringDeep);
+    expect(
+      parentFocusScope.toStringDeep(minLevel: DiagnosticLevel.info),
+      equalsIgnoringHashCodes('FocusScopeNode#00000\n'
+          '   focus: FocusNode#00000(FOCUSED)\n'),
+    );
+
+    expect(WidgetsBinding.instance.focusManager.rootScope, hasAGoodToStringDeep);
+    expect(
+      WidgetsBinding.instance.focusManager.rootScope.toStringDeep(minLevel: DiagnosticLevel.info),
+      equalsIgnoringHashCodes('FocusScopeNode#00000\n'
+          ' └─child 1: FocusScopeNode#00000\n'
+          '     focus: FocusNode#00000(FOCUSED)\n'),
+    );
+
+    parentFocusScope.setFirstFocus(childFocusScope);
```

Comment:
Added a check to make sure it got the focus. This is a little odd, since we're adding a focus scope to the focus tree before it's part of the widget hierarchy... (this part came from the existing tests).